viper/parser/stmt.py

```diff
@@ -0,0 +1,269 @@
+            ast.AugAssign: self.aug_assign,
+            ast.Break: self.parse_break,
+            ast.Return: self.parse_return,
+        }
+        stmt_type = self.stmt.__class__
+        if stmt_type in self.stmt_table:
+            self.lll_node = self.stmt_table[stmt_type]()
+        elif isinstance(stmt, ast.Name) and stmt.id == "throw":
+            self.lll_node = LLLnode.from_list(['assert', 0], typ=None, pos=getpos(stmt))
+        else:
+            raise StructureException("Unsupported statement type", stmt)
+
+    def expr(self):
+        return Stmt(self.stmt.value, self.context).lll_node
+
+    def parse_pass(self):
+        return LLLnode.from_list('pass', typ=None, pos=getpos(self.stmt))
+
+    def ann_assign(self):
+        typ = parse_type(self.stmt.annotation, location='memory')
+        varname = self.stmt.target.id
+        pos = self.context.new_variable(varname, typ)
+        return LLLnode.from_list('pass', typ=None, pos=pos)
+
+    def assign(self):
+        from .parser import (
+            parse_expr,
+            parse_variable_location,
+            make_setter,
+        )
+        # Assignment (eg. x[4] = y)
+        if len(self.stmt.targets) != 1:
+            raise StructureException("Assignment statement must have one target", self.stmt)
+        sub = parse_expr(self.stmt.value, self.context)
+        if isinstance(self.stmt.targets[0], ast.Name) and self.stmt.targets[0].id not in self.context.vars:
+            pos = self.context.new_variable(self.stmt.targets[0].id, set_default_units(sub.typ))
+            variable_loc = LLLnode.from_list(pos, typ=sub.typ, location='memory', pos=getpos(self.stmt), annotation=self.stmt.targets[0].id)
+            o = make_setter(variable_loc, sub, 'memory')
+        else:
+            target = parse_variable_location(self.stmt.targets[0], self.context)
+            if target.location == 'storage' and self.context.is_constant:
+                raise ConstancyViolationException("Cannot modify storage inside a constant function!", self.stmt.targets[0])
+            if not target.mutable:
+                raise ConstancyViolationException("Cannot modify function argument", self.stmt.targets[0])
+            o = make_setter(target, sub, target.location)
+        o.pos = getpos(self.stmt)
+        return o
+
+    def parse_if(self):
+        from .parser import (
+            parse_value_expr,
+            parse_body,
+        )
+        if self.stmt.orelse:
+            add_on = [parse_body(self.stmt.orelse, self.context)]
+        else:
+            add_on = []
+        return LLLnode.from_list(['if', parse_value_expr(self.stmt.test, self.context), parse_body(self.stmt.body, self.context)] + add_on, typ=None, pos=getpos(self.stmt))
+
+    def call(self):
+        from .parser import (
+            parse_expr,
+            pack_arguments,
+            pack_logging_data,
+            pack_logging_topics,
+            external_contract_call_stmt,
+        )
```

Comment:
Coul we add # TODO here so we can evaluate the inline imports later  (I think once we factor everything from parse.py it should be doable) ;)

Reply:
I agree 100% the inline imports are only temporary.